tts-python-phoneme-distribution.ipynb

```diff
@@ -0,0 +1,573 @@
+   "source": [
+    "# Put all the imports here for ease of use, and cleanliness of code \n",
+    "import os\n",
+    "import pprint\n",
+    "import json\n",
+    "# import nemo g2p class for grapheme to phoneme\n",
+    "from nemo_text_processing.g2p.modules import EnglishG2p\n",
+    "\n",
+    "# Bokeh\n",
+    "from bokeh.io import output_notebook, show\n",
+    "from bokeh.plotting import figure, output_file, show\n",
+    "from bokeh.io import curdoc, export_png\n",
+    "from bokeh.models import NumeralTickFormatter\n",
+    "output_notebook()\n",
+    "\n",
+    "from prettytable import PrettyTable\n",
+    "\n",
+    "# Counter \n",
+    "import collections\n",
+    "from collections import Counter\n",
+    "\n",
+    "pp = pprint.PrettyPrinter(indent =4, depth=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2cdfc8c0",
+   "metadata": {},
+   "source": [
+    "#### Get arpabet file\n",
+    "\n",
+    "Get the CMU grapheme to phoneme dictionary from [nemo](https://raw.githubusercontent.com/NVIDIA/NeMo/stable/scripts/tts_dataset_files/cmudict-0.7b_nv22.08)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6d17e187",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!wget https://raw.githubusercontent.com/NVIDIA/NeMo/stable/scripts/tts_dataset_files/cmudict-0.7b_nv22.08"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1330c959",
+   "metadata": {},
+   "source": [
+    "Create the g2p object. This object will be used throughout to convert grapheme to phoneme"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "463951e0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "g2p = EnglishG2p(phoneme_dict=\"cmudict-0.7b_nv22.08\", ignore_ambiguous_words=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "459f7ff2",
+   "metadata": {},
+   "source": [
+    "# Reference distribution:\n",
+    "\n",
+    "We will use a reference distribution that not only covers _every_ phoneme in the English language (there are 44) but also exhibits the same _frequency distribution_ as English. We can therefore use this reference distribution to compare the phoneme distribution of other datasets. This enables the identification of gaps in data, such as the need for more samples with a particular phoneme."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "931481fb",
+   "metadata": {},
+   "source": [
+    "We will use [NeMo grapheme to phoneme](https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/text_processing/g2p/g2p.html) to convert words to phonemes.  \n",
+    "List all the phonemes in g2p dict"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f1c6756b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a list of the CMU phoneme symbols\n",
+    "phonemes = {k: v[0] for k, v in g2p.phoneme_dict.items()}\n",
+    "pp.pprint(list(g2p.phoneme_dict.items())[:10])"
```

Comment:
Why do we have this magic number `[:10]`

Reply:
Its an accident. I has stayed from when I just looking at the data. I will remove this.